core/src/services/memory/backend.rs

```diff
@@ -146,11 +146,9 @@ impl typed_kv::Adapter for Adapter {
             };
```

Comment:
this `right_range` isn't precise, `{path}{char::MAX}` should include. maybe we should set right bound unbound, and filter by prefix ?

Reply:
Makes sense to me.

Reply:
I'd like to solve it in the next PR. just don't want to add too much work in this one.
ps: this PR is almost done

Reply:
Thank you very much for your effort. This issue is truly complex and time-consuming.